src/memoized.rs

```diff
@@ -129,25 +182,29 @@ where
         // first things first, let's walk over each of our previous
         // inputs and check whether they are out of date.
         if let Some(QueryState::Memoized(old_memo)) = &mut old_value {
-            if old_memo.inputs.iter().all(|old_input| {
-                !old_input.maybe_changed_since(db, old_memo.stamped_value.changed_at)
-            }) {
+            if old_memo
```

Comment:
I don't think there is any point to doing this loop unless `old_memo.value.is_some()`.